### PR TITLE
[SPARK-45723][PYTHON][CONNECT] Catalog methods avoid pandas conversion

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
Catalog methods avoid pandas conversion

### Why are the changes needed?
minor optimization:

before: arrow table -> pandas dataframe -> scalar result

after: arrow table -> scalar result

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
ci

### Was this patch authored or co-authored using generative AI tooling?
no

### CODE REVIEW

File: python/pyspark/sql/connect/catalog.py [hunk 1]
```diff
@@ -83,138 +85,125 @@ def setCurrentDatabase(self, dbName: str) -> None:
...
-            name=row.iloc[0],
-            catalog=row.iloc[1],
-            # If None, returns None.
-            namespace=None if row.iloc[2] is None else list(row.iloc[2]),
-            description=row.iloc[3],
-            tableType=row.iloc[4],
-            isTemporary=row.iloc[5],
+            name=table[0][0].as_py(),
+            catalog=table[1][0].as_py(),
+            namespace=table[2][0].as_py(),
```
Comment (zhengruifeng) [inline python/pyspark/sql/connect/catalog.py:141]: `table[2][0]` returns a `pyarrow.StringScalar` value here.
If the underlying value is None, `table[2][0]` is `<pyarrow.StringScalar: None>` and its `as_py()` is None.
So we no longer need check `None` for optional field.

```
In [11]: pdf = pd.DataFrame({"A": ["x", "y", "z", None]})

In [12]: table = pa.Table.from_pandas(pdf)

In [13]: table
Out[13]:
pyarrow.Table
A: string
----
A: [["x","y","z",null]]

In [14]: table[0][3]
Out[14]: <pyarrow.StringScalar: None>

In [15]: table[0][3].as_py()

In [16]: table[0][3].as_py() is None
Out[16]: True
```

**Comment by zhengruifeng** [comment]
cc @HyukjinKwon

**Review by HyukjinKwon — APPROVED** [review]
Nice!

**Review by HyukjinKwon — APPROVED** [review]
Nice!

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**